src/disabled_blocks.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+const lastBlockDisabledReasons: Map<string, Set<string>> = new Map();
+
+/**
+ * A change listener that enables disabled blocks when they
+ * are dragged, and re-disables them at the end of the drag.
+ *
+ * @param event Blockly event
+ */
+export function enableBlocksOnDrag(event: Blockly.Events.Abstract) {
+  // This listener only runs on Drag events that have a valid
+  // workspace and block id.
+  if (!isBlockDrag(event)) return;
+  const eventWorkspace = Blockly.common.getWorkspaceById(
+    event.workspaceId,
+  ) as Blockly.WorkspaceSvg;
+  if (!event.blockId) return;
```

Comment:
Maybe move this check before retrieving the workspace?

Reply:
Done